### DISCUSSION TITLE
Which ConnectionFactory should be used in a Spring AMPQ case?

### DISCUSSION THREAD

**Discussion by kstojanovski-work** [spring-projects/spring-amqp#2469] [Q&A] [answered]
The [documentation](https://docs.spring.io/spring-amqp/reference/html/#choosing-factory) lists three types of connection factories :
* PooledChannelConnectionFactory
* ThreadChannelConnectionFactory
* CachingConnectionFactory

Two more can be seen in the sources:
* LocalizedQueueConnectionFactory
* SimpleRoutingConnectionFactory

The documentation says that the "PooledChannelConnectionFactory" should be used for most cases.
But starting a Spring AMPQ application initializes and uses a CachingConnectionFactory. Besides that many tutorials are using the CachingConnectionFactory and the Publisher Confirms feature can be used only with this connection factory type.

Has the information about the default connection factory changed over time and the documentation needs to be adjusted or there is another reason why the PooledChannelConnectionFactory should be used for most cases?

**Comment by garyrussell** [comment, marked as answer]
`CachingConnectionFactory` was the first one available; Spring Boot auto configuration configures one.

The `PooledChannelConnectionFactory` provides similar functionality (single connection, pooled channels) but is rather simpler.

However, to use correlated publisher confirms (and returns) you have to use the `CachingConnectionFactory`.

On reflection, that documentation is too strongly worded..
